### PR TITLE
GridSample-9 specification

### PR DESCRIPTION
**Pull request by tomdol**
Based on:
https://github.com/onnx/onnx/blob/main/docs/Operators.md#GridSample
https://github.com/onnx/onnx/issues/4033
https://github.com/broune/onnx/blob/main/docs/Operators.md#GridSample

There's potentially an upcoming update of this operator but this spec is based on what was published in ONNX opset 16.
### Tickets:
 - 84456

### CODE REVIEW

File: docs/ops/image/GridSample_1.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    * `zeros` - consider values in the padding to be zeros
+    * `border` - the operator is supposed to select the nearest in-bounds pixel
+    * `reflection` - repeatedly reflect the out-of bounds value until it points to a pixel that belongs to the image
+  * **Type**: `string`
+  * **Default value**: zeros
+  * **Required**: *no*
+
+**Inputs**
+
+*   **1**: `data` - Input tensor with data to be sampled. Any supported OV data type can be used. This input is expected to be a 4-dimensional tensor with NCHW layout. **Required.**
```
Comment (mitruska) [inline docs/ops/image/GridSample_1.md:44]: Please add *Type* section and use _T_ to describe types. You can take [Interpolate-4](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/image/Interpolate_4.md) spec as reference.

File: docs/ops/image/GridSample_1.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+*   **2**: `grid` - A 4-dimensional tensor containing floating point values of normalized sampling coordinates(pairs of floats). The shape of this tensor is [N, H_out, W_out, 2] **Required.**
```
Comment (mitruska) [inline docs/ops/image/GridSample_1.md:46]: Please list another type in the Type section to describe `grid` input type.

File: docs/ops/image/GridSample_1.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+**Outputs**
+
+*   **1**: A 4-dimensional tensor with [N, C, H_out, W_out] shape. This tensor's data type is equal to the data type of the `data` input tensor. It contains the interpolated values calculated by this operator.
```
Comment (mitruska) [inline docs/ops/image/GridSample_1.md:50]: When you add _Type_ section, you can reuse type _T_ to mark type of input/output.

Comment (t-jankowski) [reply docs/ops/image/GridSample_1.md:50]:
```suggestion
*   **1**: A 4-dimensional tensor with [N, C, H_out, W_out] shape. This tensor's data type is same as the data type of the `data` input tensor. It contains the interpolated values calculated by this operator.
```

**Review by t-jankowski — APPROVED** [review]

File: docs/ops/image/GridSample_1.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
-# GridSample {#openvino_docs_ops_image_GridSample_1}
+# GridSample {#openvino_docs_ops_image_GridSample_9}
```
Comment (t-jankowski) [inline docs/ops/image/GridSample_1.md:1]: File name should change as well, shouldn't it?

**Pull request merged**